### PR TITLE
MGMT-2341 Add host hw type to host metrics

### PR DESCRIPTION
**Pull request by slaviered**
/assign @ErezAlster 
/assign @nmagnezi

### CODE REVIEW

File: internal/metrics/metricsManager.go [hunk 1]
```diff
@@ -222,10 +221,18 @@ func (m *MetricsManager) ReportHostInstallationMetrics(log logrus.FieldLogger, c
...
+			if err := json.Unmarshal([]byte(h.Inventory), &hwInfo); err == nil {
+				if hwInfo.SystemVendor != nil {
+					hwVendor = hwInfo.SystemVendor.ProductName
+				}
+			}
+			log.Infof("service Logic Host Installation Phase Seconds phase %s, vendor %s result %s, duration %f",
+				string(previousProgress.CurrentStage), hwVendor, string(phaseResult), duration)
 			m.serviceLogicHostInstallationPhaseSeconds.WithLabelValues(string(previousProgress.CurrentStage),
-				string(phaseResult), clusterVersion, h.DiscoveryAgentVersion).Observe(duration)
+				string(phaseResult), clusterVersion, h.DiscoveryAgentVersion, hwVendor).Observe(duration)
```
Comment (romfreiman) [inline internal/metrics/metricsManager.go:235]: can u provide an example? What we gonna see here?

File: internal/metrics/metricsManager.go [hunk 2]
```diff
@@ -126,7 +127,7 @@ func NewMetricsManager(registry prometheus.Registerer) *MetricsManager {
 			Help:      counterDescriptionHostInstallationPhaseSeconds,
 			Buckets: []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 20, 30, 40, 50, 60, 90, 120, 150, 180, 210, 240, 270, 300, 360, 420, 480, 540,
 				600, 900, 1200, 1500, 1800, 2100, 2400, 2700, 3000, 3300, 3600},
-		}, []string{phaseLabel, resultLabel, openshiftVersionLabel, discoveryAgentVersionLabel}),
+		}, []string{phaseLabel, resultLabel, openshiftVersionLabel, discoveryAgentVersionLabel, hwVendorLabel}),
```
Comment (romfreiman) [inline internal/metrics/metricsManager.go:130]: If you're touching this code already, can we reduce the number of buckets to 10:
https://github.com/openshift/assisted-service/pull/494/files#diff-431d86d763ee7e2b80d9e2806444ac93R110
and
https://github.com/openshift/assisted-service/pull/494/files#diff-431d86d763ee7e2b80d9e2806444ac93R128
@avishayt

**Comment by slaviered** [comment]
This is how the metrics looks like: (name of the label is under review)
service_assisted_installer_host_installation_phase_seconds_bucket{discoveryAgentVersion="",hw="VM
Virtual
host",openshiftVersion="4.5",phase="Configuring",result="Done",le="0.005"} 0

The you can query all action with phase=X and hw=Y that took more than 30
seconds for example

**Comment by romfreiman** [comment]
I think hw="VM Virtual host" is a waste of metric. If it's a vm, lets write there the hypervisor? vmware? qemu? ovirt?

**Comment by slaviered** [comment]
This label reflects the SystemVendor.ProductName :
{"manufacturer":"VMware, Inc.","product_name":"**VMware Virtual Platform**","serial_number":"VMware-56 4d 8f df 03 bd 9d 65-5f 77 1b cd 00 96 d1 1d"}}
It looks like the best field we got to reflect the host type and allows us then to construct all kind of queries (like hw=~Virtual  or hw=<all known virtual platforms>) based on the different platforms.

File: internal/metrics/metricsManager.go (hunk 2 above)
Comment (ErezAlster) [inline internal/metrics/metricsManager.go:130]: What about additional parameters such as disk info?

Comment (avishayt) [reply internal/metrics/metricsManager.go:130]: @romfreiman How about:
1, 5, 10, 30, 60, 120, 300, 600, 900, 1200, 1800

**Comment by ErezAlster** [comment]
this are the unique product name that we have at the moment in the DB:
``` Map {
  'VMware Virtual Platform' => 23,
  'PowerEdge R730xd (SKU=NotProvided;ModelName=PowerEdge R730xd)' => 4,
  'KVM' => 77,
  'PowerEdge M630 (SKU=NotProvided;ModelName=PowerEdge M630)' => 3,
  'Standard PC (Q35 + ICH9, 2009)' => 15,
  'Standard PC (i440FX + PIIX, 1996)' => 10,
  'MS-7816 (To be filled by O.E.M.)' => 1,
  'vServer' => 2,
  'PowerEdge FC430 (SKU=NotProvided;ModelName=PowerEdge FC430)' => 3,
  'BHYVE (None)' => 1,
  'RHEL' => 10,
  'To Be Filled By O.E.M.' => 3,
  'RHEL (8.2.0)' => 1,
  'QuantaGrid D52B-1U (Default string)' => 1,
  'ProLiant DL380e Gen8 (668666-421)' => 2
}
```

**Comment by ErezAlster** [comment]
and this is the combination with the manufacturer
```
'VMware, Inc.-VMware Virtual Platform' => 23,
  'Dell Inc.-PowerEdge R730xd (SKU=NotProvided;ModelName=PowerEdge R730xd)' => 4,
  'Red Hat-KVM' => 77,
  'Dell Inc.-PowerEdge M630 (SKU=NotProvided;ModelName=PowerEdge M630)' => 3,
  'QEMU-Standard PC (Q35 + ICH9, 2009)' => 15,
  'QEMU-Standard PC (i440FX + PIIX, 1996)' => 10,
  'MSI-MS-7816 (To be filled by O.E.M.)' => 1,
  'Hetzner-vServer' => 2,
  'Dell Inc.-PowerEdge FC430 (SKU=NotProvided;ModelName=PowerEdge FC430)' => 3,
  'undefined-BHYVE (None)' => 1,
  'Red Hat-RHEL' => 10,
  'Hetzner-To Be Filled By O.E.M.' => 3,
  'Red Hat-RHEL (8.2.0)' => 1,
  'Quanta Cloud Technology Inc.-QuantaGrid D52B-1U (Default string)' => 1,
  'HP-ProLiant DL380e Gen8 (668666-421)' => 2
```

**Comment by ErezAlster** [comment]
So @slaviered , Lets add two labels, one for the manufacturer and the other for product name.

**Comment by nmagnezi** [comment]
> So @slaviered , Lets add two labels, one for the manufacturer and the other for product name.

+1 , my thought exactly.
Having both the manufacturer and product name is valuable.

**Comment by nmagnezi** [comment]
@romfreiman @ErezAlster, I remember someone mentioned we are interested in collecting HD / SSD information as well.
Do we want a label for that as well?

File: subsystem/host_test.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ import (
 	"github.com/openshift/assisted-service/models"
 )
 
-var _ = Describe("Host tests", func() {
+var _ = FDescribe("Host tests", func() {
```
Comment (nmagnezi) [inline subsystem/host_test.go:20]: why?

**Comment by ErezAlster** [comment]
@slaviered is adding the HHD/SDD info as we speak :)

File: subsystem/host_test.go (hunk 3 above)
Comment (avishayt) [reply subsystem/host_test.go:20]: Well it's something we can consider using, but should standardize it

Comment (slaviered) [reply subsystem/host_test.go]: Corrected

File: internal/host/host.go [hunk 4]
```diff
@@ -505,13 +505,18 @@ func (m *Manager) reportInstallationMetrics(ctx context.Context, h *models.Host,
...
 		log.WithError(err).Errorf("not reporting installation metrics - failed to find cluster %s", h.ClusterID)
-	} else {
-		m.metricApi.ReportHostInstallationMetrics(log, cluster.OpenshiftVersion, h, previousProgress, CurrentStage)
+		return
+	}
+	//get the boot disk
+	var boot *models.Disk
+	disks, err := m.hwValidator.GetHostValidDisks(h)
+	if err == nil && len(disks) > 0 {
+		boot = disks[0]
```
Comment (ErezAlster) [inline internal/host/host.go:517]: does it mean that the first disk is the boot device?

Comment (slaviered) [reply internal/host/host.go:517]: yes. conforms to installcmd::getBootDevice

File: internal/cluster/mock_cluster_api.go [hunk 5]
```diff
@@ -6,13 +6,12 @@ package cluster
...
 import (
 	context "context"
-	reflect "reflect"
-
 	strfmt "github.com/go-openapi/strfmt"
 	gomock "github.com/golang/mock/gomock"
 	gorm "github.com/jinzhu/gorm"
 	common "github.com/openshift/assisted-service/internal/common"
 	s3wrapper "github.com/openshift/assisted-service/pkg/s3wrapper"
+	reflect "reflect"
```
Comment (nmagnezi) [inline internal/cluster/mock_cluster_api.go:14]: I suspect this moved here because you need to run `skipper make format`. checking.

Comment (nmagnezi) [reply internal/cluster/mock_cluster_api.go:14]: tested. looks like you need to correct the formatting. please run `skipper make format` and ammend.

**Pull request merged**